Leaflet.CheapLayerAt.js

```diff
@@ -22,6 +26,42 @@ L.Map.include({
 		return this._getLayerFromDOMElement(el);
 	},
 
+	getLayersAt: function(point, y) {
+		point = L.point(point, y);
+
+		// Ignore points outside the map
+		if (!this.getSize().contains(point)) { return; }
+
+		var mapPos = this._container.getBoundingClientRect();
+
+		var viewportPoint = L.point(mapPos.left, mapPos.top).add(point);
+
+		var els = this._getElementsFromPoint(viewportPoint.x, viewportPoint.y);
+		var out = [];
+		for(var i = 0; i < els.length; i += 1) {
+			var lay = this._getLayerFromDOMElement(els[i]);
+			if(lay) out.push(lay);
+		}
+		return out;
+	},
+
+	_getElementsFromPoint: function(x, y) {
+		var stack = [], el;
+		var i = 1000; // prevent from infinite loop
+		do {
+			el = document.elementFromPoint(x, y);
+			stack.push(el);
+			el.style.pointerEvents = 'none';
```

Comment:
Element's style shall **not** be overwritten without saving it somewhere first (second stack, or complicate the stack structure). Nobody knows when an element has a custom-defined `pointerEvents` style!!!